dask/dataframe/io/csv.py

```diff
@@ -329,6 +330,15 @@ def text_blocks_to_pandas(
     name = "read-csv-" + tokenize(reader, columns, enforce, head)
 
     if path:
+        path = (
+            path[0],
+            [
+                p
+                for p in path[1]
+                if os.path.basename(p) in [os.path.basename(b[1].path) for b in blocks]
```

Comment:
Can we generate this second list `[os.path.basenaem(b[1].path) for b in blocks]` once rather than once per path? Move it outside of the outer list comprehension.

Reply:
@TomAugspurger done. Thanks,